labellab-server/controller/user/userControls.js

```diff
@@ -1,146 +1,200 @@
-					data.totalImages += project.image.length
-					data.totalLabels += project.labels.length
-				})
+  User.findOne({
+    _id: req.user._id
+  })
+    .select('email')
+    .populate('project')
+    .exec(function(err, user) {
+      if (err) {
+        return res.status(400).send({
+          success: false,
+          msg: 'Unable to connect to database. Please try again.',
+          error: err
+        })
+      }
+      if (!user) {
+        return res.status(400).send({ success: false, msg: 'User not found' })
+      } else {
+        let data = {
+          totalProjects: user.project.length,
+          totalImages: 0,
+          totalLabels: 0
+        }
+        user.project.map((project, index) => {
+          data.totalImages += project.image.length
+          data.totalLabels += project.labels.length
+        })
 
-				return res.json({ success: true, body: data })
-			}
-		})
+        return res.json({ success: true, body: data })
+      }
+    })
 }
 
 exports.userUploadImage = function(req, res) {
-	if (req && req.body && req.body.image && req.body.format) {
-		const { id, email } = req.user
-		const { image, format } = req.body
-		let data = {
-			id: id,
-			email: email,
-			img: image,
-			format: format
-		}
-		let baseImg = data.img.split(',')[1]
-		let binaryData = new Buffer(baseImg, 'base64')
-		let ext = data.format.split('/')[1]
-		let updateData = { profileImage: `${data.id}.${ext}` }
-		const url = `/public/img/${updateData.profileImage}`
-		require('fs').writeFile(
-			`./public/img/${updateData.profileImage}`,
-			binaryData,
-			function(err) {
-				if (err) {
-					return res
-						.status(400)
-						.send({ success: false, msg: 'something went wrong' })
-				} else {
-					User.findOneAndUpdate(
-						{
-							email: data.email
-						},
-						{
-							profileImage:
-								'http://localhost:4000/static/img/' + updateData.profileImage
-						}
-					).exec(function(err) {
-						if (err)
-							return res.status(400).send({
-								success: false,
-								msg: 'Unable To Upload Image. Please Try Again.'
-							})
-						res.json({
-							success: true,
-							body: url,
-							msg: 'Image Uploaded Successfully.'
-						})
-					})
-				}
-			}
-		)
-	} else res.status(400).send({ success: false, msg: 'Invalid Data' })
+  if (req && req.body && req.body.image == null && req.body.format == null) {
+    const { id, email } = req.user
+    User.findOne({
+      _id: req.user.id
+    }).exec(function(err, user) {
+      if (err) {
+        return res.status(400).send({
+          success: false,
+          msg: 'Unable to connect to database. Please try again.',
+          error: err
+        })
+      }
+      if (!user) {
+        return res.status(400).send({ success: false, msg: 'User not found' })
+      } else {
+        let profileImage = user.profileImage.replace(
+          'http://localhost:4000/static/img/',
+          ''
+        )
+        require('fs').unlinkSync(
```

Comment:
Nice work and this works for already uploaded images. But if user has a profile picture imported from social account or no image at all, then it fails trying to delete the directory. Better to do a check before unlink.

Also, user updates after image saving and deletion seams duplicated redundantly. Better to refactor to make it more polished.